mora/service/itsystem.py

```diff
@@ -223,19 +228,33 @@ def get_relation_for(value, start, end):
         }
 
     def create(self, id, req):
+        systemobj = common.checked_get(req, keys.ITSYSTEM, {})
+        systemid = common.get_uuid(systemobj)
+
+        if not systemid:
+            raise ValueError('missing system uuid!')
+
+        if not util.is_uuid(systemid):
+            raise ValueError('invalid system uuid!')
+
         original = self.scope.get(
             uuid=id,
             virkningfra='-infinity',
             virkningtil='infinity',
         )
 
+        if not original:
+            raise KeyError('no such user!')
+
         rels = original['relationer'].get('tilknyttedeitsystemer', [])
 
-        rels.append(self.get_relation_for(
-            req[keys.ITSYSTEM]['uuid'],
-            common.get_valid_from(req),
-            common.get_valid_to(req),
-        ))
+        start = common.get_valid_from(req)
+        end = common.get_valid_to(req)
+
+        if start == util.negative_infinity:
+            raise ValueError('missing or invalid start date!')
```

Comment:
We never want this to happen, right? Maybe build this into `get_valid_from`?

Reply:
I actually considered that — I'll try to make the change and see if it breaks anything!